ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthEstimateGasTest.java

```diff
@@ -94,8 +94,22 @@ public void shouldReturnErrorWhenTransientTransactionProcessorReturnsEmpty() {
   }
 
   @Test
-  public void shouldReturnGasEstimateWhenTransientTransactionProcessorReturnsResultSuccess() {
-    final JsonRpcRequestContext request = ethEstimateGasRequest(callParameter());
+  public void shouldReturnErrorWhenTransientEip1559TransactionProcessorReturnsEmpty() {
+    final JsonRpcRequestContext request = ethEstimateGasRequest(eip1559TransactionCallParameter());
+    when(transactionSimulator.process(
+            eq(modifiedEip1559TransactionCallParameter()), any(OperationTracer.class), eq(1L)))
+        .thenReturn(Optional.empty());
+
+    final JsonRpcResponse expectedResponse =
+        new JsonRpcErrorResponse(null, JsonRpcError.INTERNAL_ERROR);
```

Comment:
If `process` returns `empty()`, it could be that the world state is unavailable. As I read it, that means that we're sometimes giving the user `INTERNAL_ERROR` for world state unavailability and sometimes `WORLD_STATE_UNAVAILABLE`. Is that right?

Reply:
no because the json rpc method will fail fast thanks to this check without going into the simulator.
https://github.com/hyperledger/besu/blob/83e03706b2f424afd080f18addec228d4f9aafee/ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthEstimateGas.java#L65